docs/usage/configuration/config_documentation.md

```diff
@@ -533,6 +548,21 @@ listeners:
     bind_addresses: ['::1', '127.0.0.1']
     type: manhole
 ```
+Example configuration #3:
+```yaml
+listeners:
+  # UNIX socket listener: for when Synapse is behind a reverse proxy that can utilise such.
+  #
+  # Note that x_forwarded will default to true, when using a UNIX socket. Please see
+  # https://matrix-org.github.io/synapse/latest/reverse_proxy.html.
+  #
+  - path: /var/run/synapse/main_public.sock
+    mode: 660
+    type: http
+    resources:
+      - names: [client, federation]
```

Comment:
Does it make sense to use unix sockets for anything other than `replication`?

Reply:
> Why would someone want to put `client` or `federation` over unix sockets? Are you able to have an nginx layer in front of Synapse proxy through from the unix socket?

Yes, that's it exactly. I forget you don't have the benefit of doing the other PRs for context. When using a Unix socket, you ***have*** to put a reverse proxy of some sort in front of it. Unix sockets are not capable of being served(on their own) over the interwebs. Most reverse proxies(like nginx) are the layer that handles this for you(as well as things like TLS and such).
> 
> Thanks for the background on `metrics`! Makes sense since it's served by something separate.

You're very welcome!